AsyncDisplayKit/ASViewController.mm

```diff
@@ -188,7 +202,19 @@ - (void)updateCurrentRangeModeWithModeIfPossible:(ASLayoutRangeMode)rangeMode
 
 - (ASSizeRange)nodeConstrainedSize
 {
-  CGSize viewSize = self.view.bounds.size;
+  if (AS_AT_LEAST_IOS9) {
+    CGSize viewSize = self.view.bounds.size;
+    return ASSizeRangeMake(viewSize, viewSize);
+  } else {
+    return [self _legacyConstrainedSize];
+  }
+}
+
+- (ASSizeRange)_legacyConstrainedSize
+{
+  // In modal presentation the view does not have the right bounds in iOS7 and iOS8. As workaround using the superviews
+  // view bounds
+  CGSize viewSize = (self.presentingViewController != nil) ? self.view.superview.bounds.size : self.view.bounds.size;
```

Comment:
Yep, we definitely should be defense in here. Change it and should be ready now!